src/components/pages/lessons/overlay/go-pro-cta-overlay.tsx

```diff
@@ -94,69 +97,82 @@ const GoProCtaOverlay: FunctionComponent<JoinCTAProps> = ({lesson}) => {
     if (!priceId) return
     if (!formik.values.email) return
 
-    const account = first<StripeAccount>(get(viewer, 'accounts'))
-    await track('checkout: selected plan', {
-      lesson: lesson.slug,
-      priceId: priceId,
-      location: 'lesson overlay',
-    })
+    let leaveSpinningForRedirect: boolean = false
 
-    const emailRequiresAProCheck =
-      !isEmpty(formik.values.email) && formik.values.email !== viewer?.email
+    setLoaderOn(true)
 
-    if (emailRequiresAProCheck) {
-      const {hasProAccess} = await axios
-        .post(`/api/users/check-pro-status`, {
-          email: formik.values.email,
-        })
-        .then(({data}) => data)
+    try {
```

Comment:
There is visually a lot more churn here than what is changing. This adds a `try / catch / finally` block so that the `setLoaderOn` can be reliably unset regardless of any short circuits or thrown exceptions.